### PR TITLE
Improve C wrapper coverage with boundary and error tests

### PR DESCRIPTION
**Pull request by keiji**
This PR improves the code coverage for the C wrapper (`wrapper.c`) by adding a suite of unit tests focusing on error handling and boundary conditions.

**Changes:**
*   **`test/test_wrapper.c`**:
    *   Added `malloc` interception logic to simulate memory failures.
    *   Added tests for `opj_read_from_buffer` (static function tested via direct include).
    *   Added tests for `opj_set_decode_area` failure.
    *   Added tests for partial decode pixel limits.
    *   Added tests for 0-component images.
    *   Added tests for Alpha channel detection via flags.
*   **`test/stubs.c`**:
    *   Added `stub_should_set_decode_area_succeed` flag.
    *   Added `stub_num_comps` variable.
    *   Updated mocks to respect these new controls.

**Verification:**
*   Ran `./test/run_coverage.sh` and confirmed coverage increased to ~96%.
*   All tests passed successfully.

---
*PR created automatically by Jules for task [14121585642813657781](https://jules.google.com/task/14121585642813657781) started by @keiji*

### CODE REVIEW

File: test/test_wrapper.c [hunk 1]
```diff
@@ -348,6 +355,195 @@ extern uint32_t stub_height;
...
+    assert(read == 3);
+    assert(info.offset == 3);
+    assert(buffer[0] == 1);
+
+    // 2. Partial Read at end
+    // Remaining: 2 bytes (index 3, 4)
+    read = opj_read_from_buffer(buffer, 5, &info);
+    assert(read == 2);
+    assert(info.offset == 5);
+    assert(buffer[0] == 4);
```
[the remark this answers is not in the document]
Comment (keiji) [reply test/test_wrapper.c:384]: > medium
> 
> Similar to the previous read, this assertion only checks the first of the two bytes read. To improve test robustness, it's better to check all data that was read.

File: test/test_wrapper.c [hunk 2]
```diff
@@ -348,6 +355,195 @@ extern uint32_t stub_height;
...
+    info.size = 5;
+    info.offset = 0;
+
+    uint8_t buffer[10];
+
+    // 1. Normal Read
+    OPJ_SIZE_T read = opj_read_from_buffer(buffer, 3, &info);
+    assert(read == 3);
+    assert(info.offset == 3);
+    assert(buffer[0] == 1);
```
[the remark this answers is not in the document]
Comment (keiji) [reply test/test_wrapper.c:375]: > medium
> 
> This assertion only checks the first of the three bytes read into `buffer`. To make the test more robust and ensure the function behaves as expected, it's better to verify all the data that was read.
